Sources/ClientRuntime/Telemetry/DefaultTelemetry.swift

```diff
@@ -9,6 +9,7 @@ import struct Smithy.AttributeKey
 import struct Smithy.Attributes
 import protocol Smithy.LogAgent
 import struct Smithy.SwiftLogger
+import SmithyTelemetry
```

Comment:
Move `DefaultTelemetry` from `ClientRuntime` to `SmithyTelemetry`, then `typealias` it here to prevent the breaking change.

Even though it is technically an "implementation", it has no dependencies other than `Smithy` and `SmithyTelemetry` so `SmithyTelemetry` is the right place for it.

Also, consider renaming `SmithyTelemetry` to `SmithyTelemetryAPI`, which would follow the SRA naming pattern for packages that contain interfaces but not implementations (at least not implementations with dependencies.)